boa/src/builtins/object/internal_methods.rs

```diff
@@ -352,43 +336,44 @@ impl Object {
 
     /// Helper function for property insertion.
     #[inline]
-    pub(crate) fn insert_property<N>(&mut self, name: N, p: Property)
+    pub(crate) fn insert_property<Key>(&mut self, key: Key, property: Property) -> Option<Property>
     where
-        N: Into<RcString>,
+        Key: Into<PropertyKey>,
```

Comment:
Not sure if we should use a long name here, since it could look like if we were using a type named `Key`. Maybe just `K` would be easier to spot as a generic type.

Reply:
Good point, even the standard library (`HashMap<K, V>`) does the same, I'll change it.